modules/simulation/DCSupplySimulator/main/power_supply_DCImpl.cpp

```diff
@@ -0,0 +1,117 @@
+        .max_import_current_A = static_cast<float>(config.max_current),
+        .min_import_current_A = static_cast<float>(config.min_current),
+        .max_import_power_W = static_cast<float>(config.max_power),
+        .conversion_efficiency_import = 0.85,
+        .conversion_efficiency_export = 0.9,
+    };
+    return Capabilities;
+}
+
+void power_supply_DCImpl::handle_setMode(types::power_supply_DC::Mode& value) {
+    this->mode = value;
+
+    if ((value == types::power_supply_DC::Mode::Off) || (value == types::power_supply_DC::Mode::Fault)) {
+        this->connector_voltage = 0.0;
+        this->connector_current = 0.0;
+    } else if (value == types::power_supply_DC::Mode::Export) {
+        this->connector_voltage = this->settings_connector_export_voltage;
+        this->connector_current = this->settings_connector_max_export_current;
+    } else if (value == types::power_supply_DC::Mode::Import) {
+        this->connector_voltage = this->settings_connector_import_voltage;
+        this->connector_current = this->settings_connector_max_import_current;
+    }
+
+    mod->p_main->publish_mode(value);
+}
+
+void power_supply_DCImpl::clampVoltageCurrent(double& voltage, double& current) {
+    voltage = voltage < this->config.min_voltage   ? this->config.min_voltage
+              : voltage > this->config.max_voltage ? this->config.max_voltage
+                                                   : voltage;
+
+    current = current < this->config.min_current   ? this->config.min_current
+              : current > this->config.max_current ? this->config.max_current
+                                                   : current;
+}
+
+void power_supply_DCImpl::handle_setExportVoltageCurrent(double& voltage, double& current) {
+    double temp_voltage = voltage;
+    double temp_current = current;
+
+    clampVoltageCurrent(temp_voltage, temp_current);
+
+    std::lock_guard<std::mutex> access_lock(this->power_supply_values_mutex);
+    this->settings_connector_export_voltage = temp_voltage;
+    this->settings_connector_max_export_current = temp_current;
+
+    if (this->mode == types::power_supply_DC::Mode::Export) {
+        this->connector_voltage = this->settings_connector_export_voltage;
+        this->connector_current = this->settings_connector_max_export_current;
+    }
+}
+
+void power_supply_DCImpl::handle_setImportVoltageCurrent(double& voltage, double& current) {
+    double temp_voltage = voltage;
+    double temp_current = current;
+
+    clampVoltageCurrent(temp_voltage, temp_current);
+
+    std::lock_guard<std::mutex> access_lock(this->power_supply_values_mutex);
+    this->settings_connector_import_voltage = temp_voltage;
+    this->settings_connector_max_import_current = temp_current;
+
+    if (mode == types::power_supply_DC::Mode::Import) {
+        this->connector_voltage = this->settings_connector_import_voltage;
+        this->connector_current = this->settings_connector_max_import_current;
```

Comment:
Possibly, the sign here should be inverted.
```suggestion
        this->connector_current = -this->settings_connector_max_import_current;
```